src/common/core/coretree.js

```diff
@@ -689,6 +690,9 @@ define([
 
             node = normalize(node);
 
+            //currently there is no reason to call the persist on a non-root object
+            node = getRoot(node);
+
```

Comment:
Doesn't this mean that we always get the root, but it is not necessarily normalized?